### PR TITLE
fix popup::onReactClick params

### PR DESCRIPTION
**Pull request by sumarlidason**
ref https://github.com/visgl/react-map-gl/issues/1347

### CODE REVIEW

File: src/components/popup.js [hunk 1]
```diff
@@ -179,7 +179,12 @@ function Popup(props) {
       ref={containerRef}
     >
       <div key="tip" className="mapboxgl-popup-tip" style={{borderWidth: tipSize}} />
-      <div key="content" ref={contentRef} className="mapboxgl-popup-content" onClick={onReactClick}>
+      <div
+        key="content"
+        ref={contentRef}
+        className="mapboxgl-popup-content"
+        onClick={e => onReactClick(e, {props, context})}
```
Comment (Pessimistress) [inline src/components/popup.js:186]: `onReactClick` can be `null`. You can do

```js
const thisRef = useMapControl({...props, onClick});
const {context, containerRef} = thisRef;

...

const onReactClick = useCallback(e => !context.eventManager && onClick(e, thisRef), []);

Comment (sumarlidason) [reply src/components/popup.js:186]: 🧹 thank you

File: src/components/popup.js [hunk 2]
```diff
@@ -169,7 +170,7 @@ function Popup(props) {
   );
 
   // If eventManager does not exist (using with static map), listen to React event
-  const onReactClick = context.eventManager ? null : onClick;
+  const onReactClick = useCallback(e => !context.eventManager && onClick(e, thisRef), []);
```
Comment (Pessimistress) [inline src/components/popup.js:173]: Sorry, one more thing...

```suggestion
  const onReactClick = useCallback(e => !context.eventManager && onClick(e, thisRef), [context.eventManager]);
```

**Comment by sumarlidason** [comment]
thanks for help @Pessimistress ! ❤️

**Review by Pessimistress — APPROVED** [review]

**Comment by Pessimistress** [comment]
@sumarlidason Lint is failing... You'll need to run `yarn lint fix`

**Comment by sumarlidason** [comment]
unable to repro, can you give that another go ?

image

**Comment by Pessimistress** [comment]
https://github.com/visgl/react-map-gl/pull/1348/checks?check_run_id=1874679811

**Comment by sumarlidason** [comment]
helps to pull...

g

**Pull request merged**